common/browser.go

```diff
@@ -224,8 +224,27 @@ func (b *Browser) initEvents() error {
 
 // connectionOnAttachedToTarget is called when Connection receives an attachedToTarget
 // event. Returning false will stop the event from being processed by the connection.
-func (b *Browser) connectionOnAttachedToTarget(_ *target.EventAttachedToTarget) bool {
-	return true
+func (b *Browser) connectionOnAttachedToTarget(eva *target.EventAttachedToTarget) bool {
+	// Allow connection to attach to target as the browser context is not set
+	// (hence it's the first one we have—the default), or the target is in the
+	// same browser context as the current one.
+	//
+	// This allows to attach targets to the same browser context as the current
+	// one, and to the default browser context.
+	//
+	// We don't want to hold the lock for the entire function to prevent
+	// concurrency issues with the browser context being closed while we're
+	// checking it. So, we don't use defer. This is for a future code that might
+	// extend this function, add a defer, and cause this issue to be missed.
```

Comment:
Nope, this is about keeping the lock in the shortest amount possible to prevent future regressions from modifications here. Say this method gets modified later; unintentionally, someone adds (maybe me :P) something that holds this lock longer. Since the interactions between these components are complex to figure out. I wanted to make sure that won't happen. I can put this in a closure here and use `defer`. So, the problem isn't about `defer` per se.

Reply:
How is this different from other areas of the codebase where we work with locks? Don't we fall under the same trap there too? Just trying to understand :)

Reply:
Sure :) The other areas are still prone to lock contention; we might want to fix them. I've moved this to defer instead. See b8add515.